### PR TITLE
Docs/troubleshooting

### PR DESCRIPTION
**Pull request by oblm**
#### Description
* Improves the readability and layuot of the troubleshooting doc.
* Adds a callout explaining how to workaround the "no matching compilation unit" issues that @paulbalaji and a few of our users have run into.
#### Tests
Rendered and linted in Improbadoc.
#### Documentation
This is documentation.

screencapture-localhost-8080-reference-1-0-content-troubleshooting-2018-11-16-12_43_10

### CODE REVIEW

File: docs/content/troubleshooting.md [hunk 1]
```diff
@@ -1,113 +1,119 @@
...
-**Issue and workarounds**<br/>
- You don't have the right Unity build settings enabled.<br/>
+**Cause**<br/>
+ You don't have the correct Unity build settings set.<br/>
+
+**Fix**<br/>
  In your Unity Editor:
 
-* Check what build support you have enabled (menu: **File** > **Build Settings**) and make sure you have your target build platform enabled. <br/>
+* Select **File** > **Build Settings** and make sure you have your target build platform enabled. <br/>
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: I thought we had a diagnostic tool for this? Under the `SpatialOS` menu?

File: docs/content/troubleshooting.md [hunk 2]
```diff
@@ -1,113 +1,119 @@
...
-#### Error reported
-When building workers (Unity Editor menu: **Build For Cloud** > **my_worker_name**):
+#### Error building player because build target was unsupported
+When building workers by selecting **Build For Cloud** > **my_worker_name** the following error may occur:
 **_Error building player because build target was unsupported_**</br>
 
-**Issue and workarounds**<br/>
- You don't have the right Unity build settings enabled.<br/>
+**Cause**<br/>
+ You don't have the correct Unity build settings set.<br/>
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: Its not about setting build settings, more about having build support for those platforms installed

File: docs/content/troubleshooting.md [hunk 3]
```diff
@@ -1,113 +1,119 @@
...
-_`<path to burst package>\.Runtime\bcl.exe` did not run properly!_
+#### `Runtime\bcl.exe` did not run properly!
 
-**Issue and workarounds**<br/>
-This is a benign exception that is thrown when building a worker while burst compilation is turned on. Your worker was successfully built despite this error. To remove the error message, disable Burst compilation by unchecking `Jobs > Enable Burst Compilation`
-in the Unity Editor.
+**Cause**<br/>
 
-<br/>
+This is a benign exception that is thrown when building a worker while burst compilation is turned on. Your worker was successfully built despite this error.
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: Maybe note that this isn't a problem with the GDK and burst compilation but a Unity problem with burst not fully supporting cross compilation (Win -> Linux for example)

File: docs/content/troubleshooting.md [hunk 4]
```diff
@@ -1,113 +1,119 @@
...
+The struct you defined implements [`IComponentData`](https://github.com/Unity-Technologies/EntityComponentSystemSamples/blob/132f511a0f36d2bb422fc807cb3a808ea18d7df5/Documentation/content/ecs_in_detail.md#icomponentdata) and contains [non-blittable](https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types) fields. Types that implement
 `IComponentData` may only contain blittable fields.
 
-* `bool` as well as `System.Boolean` are not blittable. Please use `Improbable.Gdk.Core.BlittableBool`, a blittable alternative
-implementation we provide, in that case.
+The most common causes of this exception are `bool` and `System.Boolean`, both of which are not blittable.
+
+**Fix**<br/>
+
+ Instead of `bool` and `System.Boolean`, please use `Improbable.Gdk.Core.BlittableBool`, a blittable alternative implementation we provide, in that case.
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: Consider rewording this sentence slightly. The `in that case` reads a bit clunky

File: docs/content/troubleshooting.md [hunk 5]
```diff
@@ -1,113 +1,119 @@
...
-These warnings are caused by code maintained by Unity over which we have no control. The warnings are benign so you can ignore them.
+#### Several warnings when opening the project for the first time
 
-<br/>
+**Cause**<br/>
+These warnings originate from code written by Unity, `com.unity.mathematics` or `com.unity.entities` for example. These warnings are benign so you can ignore them.
 
-#### Unity Editor logs warning
-Several error messages are logged after I destroyed a GameObject or ECS entity representing a SpatialOS entity.
+#### Several warnings when deleting a GameObject or ECS entity that represents a SpatialOS entity
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: I'm surprised this only throws warnings

File: docs/content/troubleshooting.md [hunk 6]
```diff
@@ -1,113 +1,119 @@
...
+**Cause**<br/>
+These warnings occur when a [GameObject]({{urlRoot}}/content/glossary#gameobject) or [ECS entity]({{urlRoot}}/content/glossary#unity-ecs-entity) that represents a SpatialOS entity is deleted before its corresponding SpatialOS entity leaves the worker's view. The GDK is designed so that the GameObjects and ECS entities that represent SpatialOS entities should only be deleted after the SpatialOS entity they represent has left that worker's view. ECS entities are deleted automatically by the GDK when their corresponding SpatialOS entities leaves that worker's view. In the case of GameObjects, the SpatialOS Runtime sends a message to the Unity worker when a SpatialOS entity leaves that worker's view, the GDK passes this message onto you. It's your responsibility to act upon these commands in your code.
 
-<br/>
+**Fix**<br/>
+When you want to delete a GameObject or ECS entity, send a [DeleteEntity world command]({{urlRoot}}/content/gameobject/world-commands.md) to delete the entity on the SpatialOS side instead.
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: Only for deleting GameObjects/ECS entities associated with a SpatialOS entity

File: docs/content/troubleshooting.md [hunk 7]
```diff
@@ -1,113 +1,119 @@
...
-  * Ensure that the [requirements]({{urlRoot}}/content/gameobject/interact-spatialos-monobehaviours) for all your fields in your MonoBehaviour are satisfied. You can verify this by examining the SpatialOS entity associated with your GameObject in the Inspector. In the Inspector, ensure that all relevant components are attached to your SpatialOS entity and that read and write access is delegated correctly for your worker.
+  * Ensure that the [requirements]({{urlRoot}}/content/gameobject/interact-spatialos-monobehaviours) for all your fields in your MonoBehaviour are satisfied. You can verify this by examining the SpatialOS entity associated with your GameObject in the [Inspector]({{urlRoot}}/content/glossary#inspector). In the Inspector, ensure that all relevant components are attached to your SpatialOS entity and that read and write access is delegated correctly for your worker.
+
+#### When opening a Unity GDK solution in Visial Studio 2017, the file has no matching compilation unit
+
+**Cause**<br/>
+You don't have the correct compilation units installed.
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: This doesn't really elucidate anything for a reader of this page

File: docs/content/troubleshooting.md [hunk 8]
```diff
@@ -1,113 +1,119 @@
...
+
+**Fix**<br/>
 
+1. Follow the Visual Studio instalation steps on our [setup page]({{urlRoot}}/setup-and-installing#set-up-your-machine)).
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: Its _probably_ to update Visual Studio to latest, but I can't guarantee that for sure.

File: docs/content/troubleshooting.md (hunk 1 above)
Comment (paulbalaji) [reply docs/content/troubleshooting.md]: yeah access it at `SpatialOS > Check build support`

File: docs/content/troubleshooting.md [hunk 9]
```diff
@@ -1,113 +1,119 @@
...
+1. Restart your computer.
```
Comment (paulbalaji) [inline docs/content/troubleshooting.md]: maybe suggest restarting unity first and if that doesn't work _then_ restart the computer

File: docs/content/troubleshooting.md (hunk 1 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: Tested this new diagnostic tool and documented. I've also added info regarding iOS and Android build support, as this tool errors when these are missing.

File: docs/content/troubleshooting.md (hunk 2 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: Clarified this.

File: docs/content/troubleshooting.md (hunk 3 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: added

File: docs/content/troubleshooting.md (hunk 4 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: fixed

File: docs/content/troubleshooting.md (hunk 5 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: I'm too busy to test this now so will leave this as is and double check what is thrown next time I deploy.

File: docs/content/troubleshooting.md (hunk 7 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: Agreed. We haven't pinned a root cause though, and there are likely several, so I'm not sure how to improve this?

File: docs/content/troubleshooting.md (hunk 6 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: good callout, added

File: docs/content/troubleshooting.md (hunk 9 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: good idea

File: docs/content/troubleshooting.md (hunk 7 above)
Comment (jamiebrynes7) [reply docs/content/troubleshooting.md]: We think the root cause is an outdated Visual Studio installation, but we've not the time or personnel to test this

File: docs/content/troubleshooting.md [hunk 10]
```diff
@@ -1,113 +1,121 @@
...
-  * You need **Windows** build support if you are developing on a Mac and want to share your game with Windows PC users. 
+  * You need **Windows** build support if you are developing on a Mac and want to share your game with Windows PC users.
+  * You need **Android** and/or **iOS** build support if you are developing for those platforms.
   * Unity gives you build support for your development machine (Windows or Mac) by default.
-  * _**In addition**_, make sure you have **Linux** build support enabled.<br/> 
- You need Linux build support because all server-workers in a cloud deployment run in a Linux environment. 
- <br/>Fix this by runing the Unity installer and selecting the appropriate option. (See [Setup and installing]({{urlRoot}}/setup-and-installing#set-up-your-machine)).
+  * _**In addition**_, make sure you have **Linux** build support enabled. You need Linux build support because all server-workers in a cloud deployment run in a Linux environment.
+
+ Fix this by runing the Unity installer and selecting the appropriate build support options during the instalation. See [Setup and installing]({{urlRoot}}/setup-and-installing#set-up-your-machine) for more information.
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: `instalation` vs `installation`. I think we prefer the American spelling

File: docs/content/troubleshooting.md [hunk 11]
```diff
@@ -1,113 +1,121 @@
...
-_`<path to burst package>\.Runtime\bcl.exe` did not run properly!_
+#### `Runtime\bcl.exe` did not run properly!
 
-**Issue and workarounds**<br/>
-This is a benign exception that is thrown when building a worker while burst compilation is turned on. Your worker was successfully built despite this error. To remove the error message, disable Burst compilation by unchecking `Jobs > Enable Burst Compilation`
-in the Unity Editor.
+**Cause**<br/>
 
-<br/>
+This is a benign exception that is thrown when building a worker while burst compilation is turned on. Your worker was successfully built despite this error. This occurs becayse Unity's burst compiler doesn't yet fully support cross compilation (Windows to Linux ,for example).
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: Last comma is in slightly wrong place

File: docs/content/troubleshooting.md [hunk 12]
```diff
@@ -1,113 +1,121 @@
...
+#### When opening a Unity GDK solution in Visial Studio 2017, the file has no matching compilation unit
+
+**Cause**<br/>
+You don't have the correct compilation units installed.
+
+**Fix**<br/>
+
+Restart your computer. We've noticed that this sometimes resolves the issue. If the issue persists:
 
+1. Follow the Visual Studio instalation steps on our [setup page]({{urlRoot}}/setup-and-installing#set-up-your-machine).
```
Comment (jamiebrynes7) [inline docs/content/troubleshooting.md]: `instalation` vs `installation`

Comment (oblm) [reply docs/content/troubleshooting.md]: fixed

File: docs/content/troubleshooting.md (hunk 11 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: fixed

File: docs/content/troubleshooting.md (hunk 10 above)
Comment (oblm) [reply docs/content/troubleshooting.md]: fixed

**Pull request merged**